torch/testing/_internal/common_distributed.py

```diff
@@ -247,8 +247,33 @@ def _run(cls, rank, test_name, file_name):
     def _join_processes(self, fn):
         timeout = get_timeout(self.id())
         start_time = time.time()
-        for p in self.processes:
-            p.join(timeout)
+        subprocess_error = False
+        while True:
+            # check to see if any subprocess exited with an error early.
+            for p in self.processes:
+                # This is the exited code processes exit with if they encountered an exception.
+                if p.exitcode == MultiProcessTestCase.TEST_ERROR_EXIT_CODE:
+                    print("Some process exited badly, terminating rest.")
+                    active_children = torch.multiprocessing.active_children()
```

Comment:
Would I be correct if I assume this only returns living sub-processes launched by the current process?